woocommerce/woocommerce
# Consistent payment and shipping options empty states

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Adds an empty state to payment methods to match shipping.

![Screenshot 2024-10-21 at 15 39 29](https://github.com/user-attachments/assets/a3720d1e-1e6e-4b9f-a29e-2dbb951c9e34)

Also:

- removes the duplicate notice from inspector for payment methods, since the content is now shown in the placeholder
- fixes the shipping method count detection which was picking up methods from deleted shipping method extensions

Closes #42090



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to WC > Settings > Shipping and delete all zones and shipping methods.
2. Go to WC > Settings > Shipping > Local pickup and ensure its off
3. Go to WC > Settings > Checkout and disable all payment methods
4. Edit the checkout block. Check the placeholder states show.
5. Check the links in the placeholder states go to the correct settings pages.
6. Re-enable shipping/payments and ensure the placeholder states are no longer shown



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
